graphrag/index/api.py

```diff
@@ -0,0 +1,161 @@
+# Copyright (c) 2024 Microsoft Corporation.
+# Licensed under the MIT License
+
+"""
+Indexing API for GraphRAG.
+
+WARNING: This API is under development and may undergo changes in future releases.
+Backwards compatibility is not guaranteed at this time.
+"""
+
+import logging
+from pathlib import Path
+
+from graphrag.config.enums import CacheType, ReportingType
+from graphrag.config.models.graph_rag_config import GraphRagConfig
+from graphrag.config.resolve_timestamp_path import resolve_timestamp_path
+
+from .cache.noop_pipeline_cache import NoopPipelineCache
+from .create_pipeline_config import create_pipeline_config
+from .emit.types import TableEmitterType
+from .progress import (
+    NullProgressReporter,
+    PrintProgressReporter,
+    ProgressReporter,
+)
+from .progress.rich import RichProgressReporter
+from .run import run_pipeline_with_config
+from .typing import PipelineRunResult
+
+
+def enable_logging(log_filepath: str | Path, verbose: bool = False) -> None:
```

Comment:
Same comment as below. I think this function can be moved to `cli.py` as `_enable_logging(...)` as it never gets called directly by the `build_index(...)` function.

Reply:
Moved the logging code to be in the config module.